### ISSUE TITLE
Support PG 13

### ISSUE THREAD

**Issue by lelit** [lelit/pglast#68]
Support for PG 13 is steadly progressing in the underlying [libpg-query](https://github.com/lfittl/libpg_query/tree/13-latest-develop).

I had an initial look at it, and I foresee a considerable effort in porting current pglast to that level. 

The most breaking thing is the introduction of a `protobuf`-based result, much richer that the `JSON`-based one: IIUC, it would make it easier to implement #60. OTOH, the Python module generated from the [definition](https://github.com/lfittl/libpg_query/blob/13-latest-develop/protobuf/pg_query.proto) is surprisingly 1.232 MB... I'm still undecided on whether it's the way to go or not.

I will upload my _wip_ branch in the following days, but **most** tests are currently failing.

**Comment by lelit** [comment]
I just pushed the status-quo in the v3 branch: all previous tests pass, but there is now a huge regressions corpus coming from the PG sources that, as expected, reveal lots of issues.

**Comment by hcheng2002cn** [comment]
@lelit what's latest status ? how can i help ? thanks

**Comment by lelit** [comment]
The v3 branch is in a good shape from my point of view, but given that it now uses the entire regression tests corpus coming from PG, there are currently slightly less than 100 failing cases.
Until next week I won't be able to switch my attention from what I'm currently focused, then I will probably ask @rdunklau, who is working on #72, about merging that PR and keep going.
So, there's nothing particularly broken, but writing/fixing printers to cover all the tests takes time. If you can lend an hand in any way, that would be great.

**Comment by hcheng2002cn** [comment]
sure, i can help on failed test case, where can i get start ? thanks

**Comment by hcheng2002cn** [comment]
@lelit I am working on some issue. one of the issue is about CAST. there is a test for:
```
create view tt20v as select * from   coalesce(1,2) as c,   collation for ('x'::text) col,   current_date as d,   localtimestamp(3) as t,   cast(1+2 as int4) as i4,   cast(1+2 as int8) as i8
```
current we only support print to (1+2)::int4 as i4, which is not valid. I manage to add feature support CAST(), however it will break  test_prettification cases. 
can i change those cast related test expected sample from :: to CAST() ?

Thanks
hong

**Comment by lelit** [comment]
Yes, you are welcome, thank you.

**Comment by lelit** [comment]
But please be sure to work on top of #72, if possible, to avoid ruining Ronan effort.

**Comment by hcheng2002cn** [comment]
@lelit yes, i am working on top of Ronan's branch.

**Comment by hcheng2002cn** [comment]
@lelit @rdunklau work progress till this weekend: 
```
=================================================================================================================================================== short test summary info ====================================================================================================================================================
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[brin.sql] - AssertionError: Statement “CREATE TABLE brintest (byteacol bytea, charcol "char", namecol name, int8col bigint, int2col smallint, int4col integer, textcol text, oidcol oid, tidcol tid, float4col real, float8col double precision, macaddrcol m...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[char.sql] - AssertionError: Statement “SELECT char 'c' = char 'c' AS true” from libpg_query/test/sql/postgres_regress/char.sql at line 2 != 'SELECT CAST((\'c\') AS char) = CAST((\'c\') AS char) AS "true"'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[foreign_key.sql] - NotImplementedError: Statement “CREATE TABLE fk (a int REFERENCES pk DEFERRABLE INITIALLY IMMEDIATE)” from libpg_query/test/sql/postgres_regress/foreign_key.sql at line 1623, could not reprint: Printer for 'CONSTR_ATTR_IMMEDIATE' of <...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[generated.sql] - NotImplementedError: Statement “ALTER TABLE gtest29 ALTER COLUMN a DROP EXPRESSION” from libpg_query/test/sql/postgres_regress/generated.sql at line 387, could not reprint: Printer for 'AT_DropExpression' of <enum 'AlterTableType'> not ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[groupingsets.sql] - NotImplementedError: Statement “select a, b, grouping(a,b), sum(v), count(*), max(v)   from gstest1 group by rollup (a,b)” from libpg_query/test/sql/postgres_regress/groupingsets.sql at line 55, could not reprint: Printer for node 'G...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[hash_func.sql] - AssertionError: Statement “SELECT v as value, hashchar(v)::bit(32) as standard,    hashcharextended(v, 0)::bit(32) as extended0,    hashcharextended(v, 1)::bit(32) as extended1 FROM   (VALUES (NULL::"char"), ('1'), ('x'), ('X'), ('p'), ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[hash_index.sql] - RuntimeError: Statement “MOVE BACKWARD 10000 FROM c” from libpg_query/test/sql/postgres_regress/hash_index.sql at line 170, could not reparse 'MOVE BACKWARD howMany=<10000> IN c': syntax error at or near "=<", at location 22
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[hs_standby_allowed.sql] - NotImplementedError: Statement “CLOSE hsc” from libpg_query/test/sql/postgres_regress/hs_standby_allowed.sql at line 92, could not reprint: Printer for node 'ClosePortalStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[identity.sql] - RuntimeError: Statement “CREATE TABLE itest3 (a smallint generated by default as identity (start with 7 increment by 5), b text)” from libpg_query/test/sql/postgres_regress/identity.sql at line 7, could not reparse 'CREATE TABLE itest3 (...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[index_including.sql] - AssertionError: Statement “ALTER TABLE tbl_include_unique1 add UNIQUE (c1, c2) INCLUDE (c3, c4)” from libpg_query/test/sql/postgres_regress/index_including.sql at line 24 != 'ALTER TABLE tbl_include_unique1 ADD UNIQUE (c1, c2)'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[index_including_gist.sql] - AssertionError: Statement “/*  * 6. EXCLUDE constraint.  */ CREATE TABLE tbl_gist (c1 int, c2 int, c3 int, c4 box, EXCLUDE USING gist (c4 WITH &&) INCLUDE (c1, c2, c3))” from libpg_query/test/sql/postgres_regress/index_includ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[interval.sql] - RuntimeError: Statement “SELECT interval(0) '1 day 01:23:45.6789'” from libpg_query/test/sql/postgres_regress/interval.sql at line 211, could not reprint: 32767
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[join.sql] - AssertionError: Statement “select * from   int8_tbl x cross join (int4_tbl x cross join lateral (select x.f1) ss)” from libpg_query/test/sql/postgres_regress/join.sql at line 1998 != 'SELECT * FROM int8_tbl AS x CROSS JOIN int4_tbl AS x CROS...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[json.sql] - AssertionError: Statement “SELECT * FROM   json_populate_record(null::record, '{"x": 776}') AS (x int, y int)” from libpg_query/test/sql/postgres_regress/json.sql at line 529 != 'SELECT * FROM json_populate_record(CAST(NULL AS record), \'{"x...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[jsonb.sql] - AssertionError: Statement “SELECT * FROM   jsonb_populate_record(null::record, '{"x": 776}') AS (x int, y int)” from libpg_query/test/sql/postgres_regress/jsonb.sql at line 649 != 'SELECT * FROM jsonb_populate_record(CAST(NULL AS record), \...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[limit.sql] - AssertionError: Statement “declare c5 cursor for select * from int8_tbl order by q1 fetch first 2 rows with ties” from libpg_query/test/sql/postgres_regress/limit.sql at line 74 != 'DECLARE c5 CURSOR FOR SELECT * FROM int8_tbl ORDER BY q1 L...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[lock.sql] - NotImplementedError: Statement “ALTER ROLE regress_rol_lock1 SET search_path = lock_schema1” from libpg_query/test/sql/postgres_regress/lock.sql at line 17, could not reprint: Printer for node 'AlterRoleSetStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[matview.sql] - NotImplementedError: Statement “REFRESH MATERIALIZED VIEW mvtest_tm” from libpg_query/test/sql/postgres_regress/matview.sql at line 20, could not reprint: Printer for node 'RefreshMatViewStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[object_address.sql] - NotImplementedError: Statement “CREATE TRANSFORM FOR int LANGUAGE SQL ( FROM SQL WITH FUNCTION prsd_lextype(internal), TO SQL WITH FUNCTION int4recv(internal))” from libpg_query/test/sql/postgres_regress/object_address.sql at line ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[partition_prune.sql] - RuntimeError: Statement “move 3 from cur” from libpg_query/test/sql/postgres_regress/partition_prune.sql at line 419, could not reparse 'MOVE FORWARD howMany=<3> IN cur': syntax error at or near "=<", at location 21
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[password.sql] - NotImplementedError: Statement “CREATE ROLE regress_passwd1 PASSWORD 'role_pwd1'” from libpg_query/test/sql/postgres_regress/password.sql at line 13, could not reprint: password
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[plpgsql.sql] - AssertionError: Statement “SELECT * FROM test_ret_set_rec_dyn(1500) AS (a int, b int, c int)” from libpg_query/test/sql/postgres_regress/plpgsql.sql at line 1542 != 'SELECT * FROM test_ret_set_rec_dyn(1500)'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[portals.sql] - RuntimeError: Statement “FETCH 2 in foo2” from libpg_query/test/sql/postgres_regress/portals.sql at line 55, could not reparse 'FETCH FORWARD howMany=<2> IN foo2': syntax error at or near "=<", at location 22
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[portals_p2.sql] - NotImplementedError: Statement “CLOSE foo13” from libpg_query/test/sql/postgres_regress/portals_p2.sql at line 72, could not reprint: Printer for node 'ClosePortalStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[privileges.sql] - RuntimeError: Statement “ALTER GROUP regress_priv_group1 ADD USER regress_priv_user4” from libpg_query/test/sql/postgres_regress/privileges.sql at line 36, could not reparse 'ALTER ROLE regress_priv_group1 ROLE regress_priv_user4': syn...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[publication.sql] - NotImplementedError: Statement “CREATE PUBLICATION testpub_default” from libpg_query/test/sql/postgres_regress/publication.sql at line 11, could not reprint: Printer for node 'CreatePublicationStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rangefuncs.sql] - RuntimeError: Statement “SELECT * FROM ROWS FROM( getrngfunc6(1) AS (rngfuncid int, rngfuncsubid int, rngfuncname text) ) WITH ORDINALITY” from libpg_query/test/sql/postgres_regress/rangefuncs.sql at line 148, could not reparse 'SELECT...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[reloptions.sql] - AssertionError: Statement “CREATE INDEX reloptions_test_idx ON reloptions_test (s) WITH (not_existing_ns.fillfactor=2)” from libpg_query/test/sql/postgres_regress/reloptions.sql at line 116 != 'CREATE INDEX reloptions_test_idx ON relop...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[returning.sql] - AssertionError: Statement “UPDATE foo SET f2 = lower(f2), f3 = DEFAULT RETURNING foo.*, f1+f3 AS sum13” from libpg_query/test/sql/postgres_regress/returning.sql at line 15 != 'UPDATE foo SET f2 = lower(f2), f3 = DEFAULT RETURNING foo.*,...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[roleattributes.sql] - NotImplementedError: Statement “CREATE ROLE regress_test_inherit WITH NOINHERIT” from libpg_query/test/sql/postgres_regress/roleattributes.sql at line 15, could not reprint: inherit
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rowsecurity.sql] - NotImplementedError: Statement “CREATE USER regress_rls_exempt_user BYPASSRLS NOLOGIN” from libpg_query/test/sql/postgres_regress/rowsecurity.sql at line 27, could not reprint: bypassrls
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rowtypes.sql] - RuntimeError: Statement “select ROW()” from libpg_query/test/sql/postgres_regress/rowtypes.sql at line 184, could not reprint: object of type 'Missing' has no len()
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rules.sql] - AssertionError: Statement “CREATE OR REPLACE RULE myrule AS ON INSERT TO ruletest_tbl DO INSTEAD INSERT INTO ruletest_tbl2 VALUES (10, 10)” from libpg_query/test/sql/postgres_regress/rules.sql at line 796 != 'CREATE RULE myrule AS ON INSERT...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[security_label.sql] - NotImplementedError: Statement “SECURITY LABEL ON TABLE seclabel_tbl1 IS 'classified'” from libpg_query/test/sql/postgres_regress/security_label.sql at line 26, could not reprint: Printer for node 'SecLabelStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[select_parallel.sql] - AssertionError: Statement “create function sp_parallel_restricted(int) returns int as   $$begin return $1; end$$ language plpgsql parallel restricted” from libpg_query/test/sql/postgres_regress/select_parallel.sql at line 2 != 'CR...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[sequence.sql] - AssertionError: Statement “CREATE UNLOGGED SEQUENCE sequence_testx” from libpg_query/test/sql/postgres_regress/sequence.sql at line 2 != 'CREATE SEQUENCE sequence_testx'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[stats_ext.sql] - RuntimeError: Statement “CREATE STATISTICS tst (unrecognized) ON x, y FROM ext_stats_test” from libpg_query/test/sql/postgres_regress/stats_ext.sql at line 40, could not reparse "CREATE STATISTICS tst ('unrecognized') ON x, y FROM ext_s...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[subscription.sql] - NotImplementedError: Statement “CREATE SUBSCRIPTION regress_testsub CONNECTION 'testconn' PUBLICATION testpub WITH (create_slot)” from libpg_query/test/sql/postgres_regress/subscription.sql at line 18, could not reprint: Printer for ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[tablesample.sql] - NotImplementedError: Statement “SELECT t.id FROM test_tablesample AS t TABLESAMPLE SYSTEM (50) REPEATABLE (0)” from libpg_query/test/sql/postgres_regress/tablesample.sql at line 7, could not reprint: Printer for node 'RangeTableSample...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[tid.sql] - NotImplementedError: Statement “REFRESH MATERIALIZED VIEW tid_matview” from libpg_query/test/sql/postgres_regress/tid.sql at line 18, could not reprint: Printer for node 'RefreshMatViewStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[tidscan.sql] - NotImplementedError: Statement “EXPLAIN (ANALYZE, COSTS OFF, SUMMARY OFF, TIMING OFF) UPDATE tidscan SET id = -id WHERE CURRENT OF c RETURNING *” from libpg_query/test/sql/postgres_regress/tidscan.sql at line 71, could not reprint: Printe...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[transactions.sql] - NotImplementedError: Statement “SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE” from libpg_query/test/sql/postgres_regress/transactions.sql at line 43, could not reprint: SET statement of kind kind=<VariableSetKi...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[triggers.sql] - RuntimeError: Statement “ALTER TRIGGER modified_a ON main_table RENAME TO modified_modified_a” from libpg_query/test/sql/postgres_regress/triggers.sql at line 351, could not reparse 'ALTER TRIGGER main_table RENAME TO modified_modified_a...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[truncate.sql] - RuntimeError: Statement “CREATE TABLE truncate_b (id int GENERATED ALWAYS AS IDENTITY (START WITH 44))” from libpg_query/test/sql/postgres_regress/truncate.sql at line 205, could not reparse 'CREATE TABLE truncate_b ( id integer GENERATE...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[type_sanity.sql] - AssertionError: Statement “SELECT p1.oid, p1.typname, p1.typalign, p2.typname, p2.typalign FROM pg_type as p1      LEFT JOIN pg_range as r ON rngtypid = p1.oid      LEFT JOIN pg_type as p2 ON rngsubtype = p2.oid WHERE p1.typtype = 'r'...
=============================================================================================================================================== 45 failed, 825 passed in 45.18s ================================================================================================================================================
```

**Comment by lelit** [comment]
:+1: :clap:

**Comment by hcheng2002cn** [comment]
@lelit how about optional ? I am working on alias, which AS can be optional. If i want to pass case for test_pg_regress_corpus, it will break test_prettification. Can i modify sql test sample in test_prettification ?

**Comment by lelit** [comment]
hcheng2002cn ***@***.***> writes:

> @lelit how about optional ? I am working on alias, which AS can be optional.
> If i want to pass case for test_pg_regress_corpus, it will break
> test_prettification. Can i modify sql test sample in test_prettification ?

I'm not sure I understand what you are asking: usually I prefer printing back
optional keywords, but I don't see how it would break the prettification
test...

Can you clarify, pointing to the exact statement?
-- 
nickname: Lele Gaifax | Quando vivrò di quello che ho pensato ieri
real: Emanuele Gaifas | comincerò ad aver paura di chi mi copia.
***@***.***  |                 -- Fortunato Depero, 1929.

**Comment by hcheng2002cn** [comment]
> hcheng2002cn ***@***.***> writes:
> @lelit how about optional ? I am working on alias, which AS can be optional. If i want to pass case for test_pg_regress_corpus, it will break test_prettification. Can i modify sql test sample in test_prettification ?
> I'm not sure I understand what you are asking: usually I prefer printing back optional keywords, but I don't see how it would break the prettification test... Can you clarify, pointing to the exact statement?

Never mind, i passed this.

**Comment by hcheng2002cn** [comment]
Current status **16 failed, 854 passed**:
FAILED tests/test_printers_prettification.py::test_prettification[test_printers_prettification/dml/select.sql-118-SELECT pe.id\nFROM table1 as pe\nINNER JOIN table2 AS pr ON pe.project_id = pr.id\nLEFT JOIN (table3 AS cp INNER JOIN table4 AS c ON cp.company_id = c.id)\n       ON cp.person_id = pe.id\n=\nSELECT pe.id\nFROM table1 AS pe\n     INNER JOIN table2 AS pr ON pe.project_id = pr.id\n     LEFT JOIN table3 AS cp\n               INNER JOIN table4 AS c ON cp.company_id = c.id\n        ON cp.person_id = pe.id]
FAILED tests/test_printers_prettification.py::test_prettification[test_printers_prettification/dml/select.sql-143-select c_id\nfrom (select c_id, row_number() over (order by c_d_id) as rn,  count(*) over() max_rn\n      from customer where c_d_id=5) t\nwhere rn = (select floor(random()*(max_rn))+1)\n=\nSELECT c_id\nFROM (SELECT c_id\n           , row_number() OVER (ORDER BY c_d_id) AS rn\n           , count(*) OVER () AS max_rn\n      FROM customer\n      WHERE c_d_id = 5) AS t\nWHERE rn = (SELECT (floor((random() * max_rn)) + 1))]
FAILED tests/test_printers_prettification.py::test_prettification[test_printers_prettification/dml/select.sql-156-select a.* from a left join (select distinct id from b) as b on a.id = b.id\n=\nSELECT a.*\nFROM a\n     LEFT JOIN (SELECT DISTINCT id\n                FROM b) AS b ON a.id = b.id] - AssertionError: /Users/hcheng/Do...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[brin.sql] - AssertionError: Statement “CREATE TABLE brintest (byteacol bytea, charcol "char", namecol name, int8col bigint, int2col smallint, int4col integer, textcol text, oidcol oid, tidcol tid, float4col real, float8col double precision, macaddrcol macaddr, inet...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[char.sql] - AssertionError: Statement “SELECT char 'c' = char 'c' AS true” from libpg_query/test/sql/postgres_regress/char.sql at line 2 != 'SELECT CAST((\'c\') AS char) = CAST((\'c\') AS char) AS "true"'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[hash_func.sql] - AssertionError: Statement “SELECT v as value, hashchar(v)::bit(32) as standard,    hashcharextended(v, 0)::bit(32) as extended0,    hashcharextended(v, 1)::bit(32) as extended1 FROM   (VALUES (NULL::"char"), ('1'), ('x'), ('X'), ('p'), ('N')) x(v) ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rowtypes.sql] - RuntimeError: Statement “select ROW()” from libpg_query/test/sql/postgres_regress/rowtypes.sql at line 184, could not reprint: object of type 'Missing' has no len()
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[rules.sql] - AssertionError: Statement “create rule rr as on update to rule_t1 do instead UPDATE rule_dest trgt   SET (f2[1], f1, tag) = (SELECT new.f2, new.f1, 'updated'::varchar)   WHERE trgt.f1 = new.f1 RETURNING new.*” from libpg_query/test/sql/postgres_regress...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[security_label.sql] - NotImplementedError: Statement “SECURITY LABEL ON TABLE seclabel_tbl1 IS 'classified'” from libpg_query/test/sql/postgres_regress/security_label.sql at line 26, could not reprint: Printer for node 'SecLabelStmt' is not implemented yet
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[select_parallel.sql] - AssertionError: Statement “create function sp_parallel_restricted(int) returns int as   $$begin return $1; end$$ language plpgsql parallel restricted” from libpg_query/test/sql/postgres_regress/select_parallel.sql at line 2 != 'CREATE FUNCTIO...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[sequence.sql] - AssertionError: Statement “CREATE UNLOGGED SEQUENCE sequence_testx” from libpg_query/test/sql/postgres_regress/sequence.sql at line 2 != 'CREATE SEQUENCE sequence_testx'
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[stats_ext.sql] - RuntimeError: Statement “CREATE STATISTICS tst (unrecognized) ON x, y FROM ext_stats_test” from libpg_query/test/sql/postgres_regress/stats_ext.sql at line 40, could not reparse "CREATE STATISTICS tst ('unrecognized') ON x, y FROM ext_stats_test": ...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[subscription.sql] - RuntimeError: Statement “CREATE SUBSCRIPTION regress_testsub CONNECTION 'dbname=regress_doesnotexist' PUBLICATION foo, testpub, foo WITH (connect = false)” from libpg_query/test/sql/postgres_regress/subscription.sql at line 25, could not reparse...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[transactions.sql] - RuntimeError: Statement “SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE” from libpg_query/test/sql/postgres_regress/transactions.sql at line 43, could not reparse 'SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, , ': syntax er...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[triggers.sql] - RuntimeError: Statement “ALTER TRIGGER modified_a ON main_table RENAME TO modified_modified_a” from libpg_query/test/sql/postgres_regress/triggers.sql at line 351, could not reparse 'ALTER TRIGGER main_table RENAME TO modified_modified_a': syntax er...
FAILED tests/test_printers_roundtrip.py::test_pg_regress_corpus[type_sanity.sql] - AssertionError: Statement “SELECT p1.oid, p1.typname, p1.typalign, p2.typname, p2.typalign FROM pg_type as p1      LEFT JOIN pg_range as r ON rngtypid = p1.oid      LEFT JOIN pg_type as p2 ON rngsubtype = p2.oid WHERE p1.typtype = 'r' AND     (p1...
================================================================================================================================================ 16 failed, 854 passed in 66.32s (0:01:06) =================================================================================================================================================

**Comment by lelit** [comment]
Wow, kudos!

**Comment by lelit** [comment]
I released an intermediary v3.0dev0, I consider this done.
Thank you again!
